### PR TITLE
Add options to enforce autoformatter version

### PR DESCRIPTION
**Pull request by joshessman-llnl**
This PR allows projects to specify the autoformatter version they require, which could prevent situations where different tool versions could result in inconsistent changes being made between users.

### CODE REVIEW

File: cmake/SetupCodeChecks.cmake [hunk 1]
```diff
@@ -654,7 +681,17 @@ macro(blt_add_uncrustify_target)
...
+
+        if(BLT_REQD_UNCRUSTIFY_VER)
+            # The user may only specify a part of the version (e.g. just the maj ver)
+            # so check for substring
+            string(FIND ${_uncrustify_version} ${BLT_REQD_UNCRUSTIFY_VER} VERSION_MATCHES)
+            if (VERSION_MATCHES EQUAL -1)
+                message(FATAL_ERROR "blt_add_uncrustify_target: uncrustify ${BLT_REQD_UNCRUSTIFY_VER} is required, found ${_uncrustify_version}")
+            endif()
+        endif()
+    endif()
```
Comment (davidbeckingsale) [inline cmake/SetupCodeChecks.cmake:694]: This diff looks like you have added an unmatched endif? If not, can you fix the indenting for it.

**Review by davidbeckingsale — CHANGES_REQUESTED** [review]
One quick comment.

**Review by kennyweiss — APPROVED** [review]
Thanks @joshessman-llnl !

File: cmake/SetupCodeChecks.cmake [hunk 2]
```diff
@@ -552,6 +563,21 @@ macro(blt_add_clangformat_target)
         set(_wd ${CMAKE_CURRENT_SOURCE_DIR})
     endif()
 
+    # If a required version was set, check it
+    if(BLT_REQD_CLANGFORMAT_VER)
+        execute_process(COMMAND ${CLANGFORMAT_EXECUTABLE} --version
```
Comment (kennyweiss) [inline cmake/SetupCodeChecks.cmake:568]: Addresses #397

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ The project release numbers follow [Semantic Versioning](http://semver.org/spec/
   parameter of blt_add_executable
 - Added variable BLT_CMAKE_IMPLICIT_LINK_DIRECTORIES_EXCLUDE that will be used to filter
   implicit link directories for all languages
+- Added user option for enforcing specific versions of autoformatters
```
[the remark this answers is not in the document]
Comment (joshessman-llnl) [reply RELEASE-NOTES.md]: Thanks for taking a look at this!  I think this should support specific versions as well as the version checks make sure that the found version "starts with" the required version.  For example, the clang-format on my machine is `10.0.0-4ubuntu1`, so a project could set `BLT_REQUIRED_CLANGFORMAT_VER` to any of `10`, `10.0`, `10.0.0`, etc and have the check succeed.

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -18,6 +18,8 @@ The project release numbers follow [Semantic Versioning](http://semver.org/spec/
   parameter of blt_add_executable
 - Added variable BLT_CMAKE_IMPLICIT_LINK_DIRECTORIES_EXCLUDE that will be used to filter
   implicit link directories for all languages
+- Added user option for enforcing specific versions of autoformatters - the new options are
+  ``BLT_REQUIRED_ASTYLE_VER``, ``BLT_REQUIRED_CLANGFORMAT_VER``, and ``BLT_REQUIRED_UNCRUSTIFY_VER``
```
Comment (white238) [inline RELEASE-NOTES.md:22]: I think on a whole (though I'm not saying we are perfect), BLT generally doesn't shorten variables names.  Can you change this to VERSION not VER?

File: docs/api/code_check.rst [hunk 5]
```diff
@@ -80,6 +80,17 @@ are out of compliance with your code formatting and a `style` build target that
 modify your source files.  It also creates smaller child build targets that follow the pattern
 `<PREFIX>_<astyle|clangformat|uncrustify>_<check|style>`.
 
+If a particular version of a code formatting tool is required, you can configure BLT to enforce
```
Comment (white238) [inline docs/api/code_check.rst:83]: Thanks for doing the documentation!

**Review by white238 — APPROVED** [review]

**Review by davidbeckingsale — APPROVED** [review]

**Pull request merged**